integration_tests/gruntz_demo3.py

```diff
@@ -0,0 +1,184 @@
+from lpython import S, str
+from sympy import Symbol, Pow, sin, oo, pi, E, Mul, Add
+
+def mrv(e: S, x: S) -> list[S]:
+    """
+    Calculate the MRV set of the expression.
+
+    Examples
+    ========
+
+    >>> mrv(log(x - log(x))/log(x), x)
+    {x}
+
+    """
+
+    if e == x:
+        list1: list[S] = [x]
+        return list1
+    if e.func == Mul or e.func == Add:
+        a: S = e.args[0]
+        b: S = e.args[1]
+        ans1: list[S] = mrv(a, x)
+        ans2: list[S] = mrv(b, x)
+        return mrv_max(mrv(a, x), mrv(b, x), x)
+    if e.func == Pow:
+        base: S = e.args[0]
+        list3: list[S] = mrv(base, x)
+        return list3
+    if e.is_Function:
+        return reduce(lambda a, b: mrv_max(a, b, x), (mrv(a, x) for a in e.args))
+    raise NotImplementedError(f"Can't calculate the MRV of {e}.")
+
+def mrv_max(f: list[S], g: list[S], x: S) -> list[S]:
+    """Compute the maximum of two MRV sets.
+
+    Examples
+    ========
+
+    >>> mrv_max({log(x)}, {x**5}, x)
+    {x**5}
+
+    """
+
+    if len(f) == 0:
+        return g
+    elif len(g) == 0:
+        return f
+    else:
+        return f | g
+
+
+def sign(e: S) -> S:
+    """
+    Returns the complex sign of an expression:
+
+    Explanation
+    ===========
+
+    If the expression is real the sign will be:
+
+        * $1$ if expression is positive
+        * $0$ if expression is equal to zero
+        * $-1$ if expression is negative
+    """
+
+    if e.is_positive:
+        return S(1)
+    elif e == S(0):
+        return S(0)
+    else:
+        return S(-1)
+
+def signinf(e: S, x : S) -> S:
+    """
+    Determine sign of the expression at the infinity.
+
+    Returns
+    =======
+
+    {1, 0, -1}
+        One or minus one, if `e > 0` or `e < 0` for `x` sufficiently
+        large and zero if `e` is *constantly* zero for `x\to\infty`.
+
+    """
+
+    if not e.has(x):
+        return sign(e)
+    if e == x:
+        return S(1)
+    if e.func == Pow:
+        base: S = e.args[0]
+        if signinf(base, x) == S(1):
+            return S(1)
+
+def mrv_leadterm(e: S, x: S) -> list[S]:
+    """
+    Compute the leading term of the series.
+
+    Returns
+    =======
+
+    tuple
+        The leading term `c_0 w^{e_0}` of the series of `e` in terms
+        of the most rapidly varying subexpression `w` in form of
+        the pair ``(c0, e0)`` of Expr.
+
+    Examples
+    ========
+
+    >>> leadterm(1/exp(-x + exp(-x)) - exp(x), x)
+    (-1, 0)
+
+    """
+
+    #w = Dummy('w', real=True, positive=True)
+    w: S = Symbol('w')
+    #e = rewrite(e, x, w)
+    coeff_exp_list: list[S] = [S(2), S(3)]
+    #return e.leadterm(w)
+    return coeff_exp_list
```

Comment:
Blocker 3:

We would need a leadterm function here. The leadterm for an expression `e` would end up in the form `c*x**n`.
If using sympy, we do
```
>>> 2*x**3
2*x**3
>>> _.as_coeff_exponent(x)
(2, 3)
```
But we might not have the `as_coeff_exponent` method. What we have is `args`. So we can maybe try the following workaround 
```
>>> 2*x**3
2*x**3
>>> _.args
(2, x**3)
>>> 2*x**3
2*x**3
>>> _.args
(2, x**3)
>>> c = _[0]
>>> 2*x**3/c
x**3
>>> _.args
(x, 3)
>>> n = _[1]
>>> c, n
(2, 3)
```